zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+from datetime import datetime
+
+import flask
+from flask import request
+
+from .utils.json_result import json_result
+from .utils.route_wrappers import cross_domain, with_session
+from . import api
+import zeeguu
+from zeeguu.model import User, Cohort, UserActivityData, Session, Language
+import sqlalchemy
+from flask import jsonify
+import json
+import random
+import datetime
+
+# class function wrapper
+def class_function_checker(class_id):
+    from zeeguu.model import TeacherCohortMap
+    link = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+    for l in link:
+        if l.user_id == flask.g.user.id:
+            return True
+    return False
+
+
+# Takes user_id and returns user.name that corresponds
+#@api.route("/get_user_name/<id>", methods=["GET"])
+@with_session
+def get_user_name(id):
+    user = User.query.filter_by(id=id).one()
+    return user.name
+
+
+# Checks to see if user has a valid active session
+@api.route("/has_session", methods=["GET"])
+def has_session():
+    try:
+        session_id = int(flask.request.args['session'])
+        session = Session.query.filter_by(id=session_id).one()
+        if session is None:
+            #print("no session exists")
+            return jsonify(0)
+
+        return jsonify(1)
+    except:
+        #print("exception called")
+        return jsonify(0)
+
+
+# Checks to see if user has permissions to access a certain class.
+@api.route("/get_class_permissions/<id>", methods=["GET"])
+@with_session
+def get_class_permissions(id):
+    if(class_function_checker(id)):
+        return jsonify(1)
+    return jsonify(0)
+
+@api.route("/get_user_permissions/<id>", methods=["GET"])
+@with_session
+def get_user_permissions(id):
+    try:
+        user = User.query.filter_by(id=id).one()
+        return get_class_permissions(user.cohort_id)
+    except:
+        flask.abort(401)
+
+
+# Asking for a nonexistant cohort will cause .one() to crash!
+# Takes cohort_id and returns all users belonging to that cohort
+@api.route("/get_users_from_class/<id>", methods=["GET"])
+@with_session
+def get_users_from_class(id):
+    if(not class_function_checker(id)):
+        flask.abort(401)
+    c = Cohort.query.filter_by(id=id).one()
+    if not c is None:
```

Comment:
ca can not be None. 
either it returns something, or throws exception.
thus there should be a try:/exxcept: here.
or?

Reply:
Instead of c is None, now use try and catch the exception.